heclib/heclib_c/src/Public/zpdRetrieve.c

```diff
@@ -705,8 +706,272 @@ int zpdRetrieve(long long *ifltab, zStructPairedData *pds, int retrieveSizeFlag)
+					if (indElev) {
+						//-----------------------------//	
+						// determine the offset to use //
+						//-----------------------------//	
+						double offset;
+						switch(ivertical_datum) {
+							case IVERTICAL_DATUM_NAVD88 :
+								offset = vdi->offsetToNavd88;
+								break;
+							case IVERTICAL_DATUM_NGVD29 :
+								offset = vdi->offsetToNgvd29;
+								break;
+							default :
+								if(!strcmp(cvertical_datum, vdi->nativeDatum) || !strcmp(cvertical_datum, CVERTICAL_DATUM_OTHER)) {
+									offset = 0;
+								}
+								else {
+									offset = UNDEFINED_VERTICAL_DATUM_VALUE;
+								}
+								break;
+						}
+						if (offset != 0.) {
+							offset = getOffset(offset, vdi->unit, pds->unitsIndependent);
+							if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
+								if (indElev) {
+									sprintf(
+										errmsg,
+										"\nData unit (%s) and/or offset unit (%s) is invalid for vertical datum conversion.\n"
+										"Conversion to datum '%s' could not be performed.\n"
+										"No data stored.",
+										pds->unitsIndependent,
+										vdi->unit,
+										cvertical_datum);
+									zmessage(ifltab, errmsg);
+								}
+								else {
+									if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
+										sprintf(
+											errmsg,
+											"\nData unit (%s) and/or offset unit (%s) is invalid for vertical datum conversion.\n"
+											"Conversion to datum '%s' could not be performed.\n"
+											"No data stored.",
+											pds->unitsDependent,
+											vdi->unit,
+											cvertical_datum);
+										zmessage(ifltab, errmsg);
+									}
+								}
+							}
+							else {
+								//------------------------------//
+								// add the offset to the values //
+								//------------------------------//
+								if (pds->doubleOrdinates) {
+									for (int i = 0; i < pds->numberOrdinates; ++i) {
+										pds->doubleOrdinates[i] += offset;
+									}
+								}
+								else if (pds->floatOrdinates) {
+									for (int i = 0; i < pds->numberOrdinates; ++i) {
+										pds->floatOrdinates[i] += offset;
+									}
+								}
+							}
+						}
+					}
+					if (depElev) {
+						//-----------------------------//	
+						// determine the offset to use //
+						//-----------------------------//	
+						double offset;
+						switch(ivertical_datum) {
+							case IVERTICAL_DATUM_NAVD88 :
+								offset = vdi->offsetToNavd88;
+								break;
+							case IVERTICAL_DATUM_NGVD29 :
+								offset = vdi->offsetToNgvd29;
+								break;
+							default :
+								if(!strcmp(cvertical_datum, vdi->nativeDatum) || !strcmp(cvertical_datum, CVERTICAL_DATUM_OTHER)) {
+									offset = 0;
+								}
+								else {
+									offset = UNDEFINED_VERTICAL_DATUM_VALUE;
+								}
+								break;
+						}
+						if (offset != 0.) {
+							offset = getOffset(offset, vdi->unit, pds->unitsDependent);
+							if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
+								if (indElev) {
```

Comment:
The following if block is redundant. We took care of the indElev case above and are in a block where depElev = TRUE